stdlib/dune

```
@@ -22,7 +22,11 @@
  (flags (:standard -w -9 -nolabels))
  (preprocess
    (per_module
-     ((action (run awk -f %{dep:expand_module_aliases.awk} %{input-file}))
+    ((action
+      (progn
+       (run sed -i s/%atomic_load/%identity/ %{input-file})
+       (run sed -i s/%atomic_cas/%obj_set_field/ %{input-file})
+       (run awk -f %{dep:expand_module_aliases.awk} %{input-file})))
```

There's definitely no need to call `sed` twice, but I'm also not sure that Dune expects this file to being altered in this way (i.e. having a pre-processor modify the input file - it's meant to pre-process it to stdout).  I would expect that changing the `lang dune` in `dune-project` to 2.7 and then:

```suggestion
    ((action
      (pipe-stdout
        (run sed s/%atomic_load/%identity/;s/%atomic_cas/%obj_set_field/ %{input-file})
        (run awk -f %{dep:expand_module_aliases.awk})))
```
should work better.

All that said, I'm not clear why this change is necessary at all!

---

Thanks for suggestion. As you said, using `pipe-stdout` can be adopted only after changing `dune-project` to 2.7.
For the reason why this change is needed, I believe that as of #11038 the primitives needing awful hacks are introduced in `stdlib.ml` also.